rskj-core/src/main/java/co/rsk/mine/ForkDetectionDataCalculator.java

```diff
@@ -63,8 +64,8 @@
         byte[] commitToParentsVector = buildCommitToParentsVector(mainchainBlockHeaders);
         System.arraycopy(commitToParentsVector, 0, forkDetectionData, 0, 7);
 
-        short numberOfUncles = getNumberOfUncles(mainchainBlockHeaders);
-        forkDetectionData[7] = ByteBuffer.allocate(2).putShort(numberOfUncles).array()[1];
+        Uint8 numberOfUncles = getNumberOfUncles(mainchainBlockHeaders);
+        forkDetectionData[7] = numberOfUncles.asByte();
```

Comment:
Aren't you re-implementing encoding here? I would suggest having an `encode(array, offset)` overload for this use case (and many more to come!) and remove `asByte`.

Reply:
The `asByte` method is what I need for this case and it makes total sense to have it since Uint8 is just a byte wrapper.